var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,811 @@
+        self.setup_env_vars(env)
+
+    def setup_run_environment(self, env):
+        self.setup_env_vars(env)
+        env.prepend_path(
+            "PATH", join_path(self.prefix.share, "chapel", self._output_version_short, "util")
+        )
+
+    @property
+    @llnl.util.lang.memoized
+    def get_package_modules(self):
+        test_modules = set()
+        for module in self.spec.variants["package_modules"].value:
+            if module == "all":
+                for m in self.package_module_dict.keys():
+                    test_modules.add(m)
+            elif module != "none":
+                test_modules.add(module)
+        return test_modules
+
+    @property
+    @llnl.util.lang.memoized
+    def _output_version_long(self):
+        if str(self.spec.version).lower() == "main":
+            return "2.1.0"
+        spec_vers_str = str(self.spec.version.up_to(3))
+        return spec_vers_str
+
+    @property
+    @llnl.util.lang.memoized
+    def _output_version_short(self):
+        if str(self.spec.version).lower() == "main":
+            return "2.1"
+        spec_vers_str = str(self.spec.version.up_to(2))
+        return spec_vers_str
+
+    def test_version(self):
+        """Perform version checks on selected installed package binaries."""
+        expected = f"version {self._output_version_long}"
+        exes = ["chpl"]
+
+        if self.spec.variants["chpldoc"].value:
+            exes.append("chpldoc")
+
+        for exe in exes:
+            reason = f"ensure version of {exe} is {self._output_version_long}"
+            with test_part(self, f"test_version_{exe}", purpose=reason):
+                path = join_path(self.prefix.bin, exe)
+                if not os.path.isfile(path):
+                    raise SkipTest(f"{path} is not installed")
+                prog = which(path)
+                if prog is None:
+                    raise RuntimeError(f"Could not find {path}")
+                output = prog("--version", output=str.split, error=str.split)
+                assert expected in output
+
+    def check(self):
+        # TODO: This is here because it's hard to have our make test target work
+        # with the spack test framework. Our make test target relies on using
+        # `start_test` and it really expects to be ran from the source directory
+        # but spack creates a test cache directory and runs the tests from there.
+        # we could conceivably just copy most everything over to the cache directory,
+        # but the other issue is that these tests take a long time to run and
+        # it's not likely users are going to want to wait 30 minutes or maybe some
+        # number of hours for the tests to run. So we're going to skip this for now
+        # and just rely on minimal hello world and version number tests.
+        # We can't just run make check here because chpl isn't yet installed
+        # and we need it to be in the PATH for make check to work
+        pass
```

Comment:
(minor) It might be worth reorganizing/shortening:

* By default all `AutotoolsPackage`'s run `make check` and `make test` as part of the `check` phase
* You are overriding it to skip those because
  * `make check` doesn't work at build time
  * `make test` is expensive